### DISCUSSION TITLE
Lang Support status

### DISCUSSION THREAD

**Discussion by JarbasAl** [OpenVoiceOS/ovos-buildroot#115] [General]
We want to add a language selection step in our initial setup skill

work has started for building a dynamic list of config options per language based on installed plugins, however we still need to select the languages we officially support

Checklist:
- [ ] lingua_franca parse utils
- [ ] lingua_franca format utils
- [ ] ovos default skills support that lang
- [ ] we have a online STT (high accuracy) option for that lang
- [ ] we have a offline STT option for that lang
- [ ] we have a online  (high quality) male TTS option for that lang
- [ ] we have a offline male TTS option for that lang
- [ ] we have a online  (high quality) female TTS option for that lang
- [ ] we have a offline female TTS option for that lang

Skills/Plugins checklist:
- [ ] [skill-ovos-homescreen](https://github.com/OpenVoiceOS/skill-ovos-homescreen)
- [ ] [skill-ovos-timer](https://github.com/OpenVoiceOS/skill-ovos-timer)
- [ ] [skill-ovos-weather](https://github.com/OpenVoiceOS/skill-ovos-weather)
- [ ] .... TODO full list

The aim of this discussion is to compile a list of tasks for each language we want to support, as well as to define the minimal set of plugins for that language

Once we have the checklist above complete we can then add that language to a selection menu

Want your language supported? help us out!

**Comment by JarbasAl** [comment]
English Checklist:

- [x] lingua_franca parse utils
- [x] lingua_franca format utils
- [x] ovos default skills support that lang
- [x] we have a online STT (high accuracy) option for that lang
    - google
- [x] we have a offline STT option for that lang
    - vosk, scribosermo, silero
- [x] we have a online  (high quality) male TTS option for that lang
    - mimic2, mimic3, marytts
- [x] we have a offline male TTS option for that lang
    - mimic, espeak
- [x] we have a online  (high quality) female TTS option for that lang
    - coqui / mozilla / glados / larynx / mimic3 / marytts
- [x] we have a offline female TTS option for that lang
   - pico / espeak
   
   
 Skills/Plugins checklist:
- [x] [skill-ovos-homescreen](https://github.com/OpenVoiceOS/skill-ovos-homescreen)
- [x] [skill-ovos-timer](https://github.com/OpenVoiceOS/skill-ovos-timer)
- [x] [skill-ovos-weather](https://github.com/OpenVoiceOS/skill-ovos-weather)
- [ ] .... TODO full list


Minimal plugin recommendations:
   
offline:
   - male TTS - mimic1/3 depending on hardware
   - female TTS - pico/mimic3 depending on hardware
   - STT - vosk
  
online:
   - male TTS - mimic3 (marytts plugin)
   - female TTS - coqui / mimic3  (marytts plugin)
   - STT - google (ovos proxy)

**Comment by NeonDaniel** [comment]
Related, there are some helper methods in neon-utils to collect supported languages at runtime https://github.com/NeonGeckoCom/neon-skill-utils/pull/364/files#diff-4265d4d0fa4cc58a5f4677f75025dac405ceb724592c6bdb03c502bdb24f3cc7R51

A few considerations:
- Offline options may differ between ARM and x86
- Different locales are often separate voices in TTS and may have different STT models too (en-gb isn't the same as en-us)

**Comment by AIIX** [comment]
I am adding a checklist based on the above for each language that has some initial support planned for below

**Comment by AIIX** [comment]
French Checklist:

- [x] lingua_franca parse utils
- [x] lingua_franca format utils
- [ ] ovos default skills support that lang
- [x] we have a online STT (high accuracy) option for that lang
- [x] we have a offline STT option for that lang
- [x] we have a online  (high quality) male TTS option for that lang
- [ ] we have a offline male TTS option for that lang
- [x] we have a online  (high quality) female TTS option for that lang
- [x] we have a offline female TTS option for that lang
      
Minimal plugin recommendations:
offline:
   - male TTS - ?
       - espeak works but sucks
   - female TTS - pico
   - STT - vosk
  
online:
   - male TTS - larynx - fr-fr/tom-glow_tts
   - female TTS - larynx - fr-fr/siwis-glow_tts
   - STT - google (ovos proxy)

**Comment by AIIX** [comment]
Spanish Checklist:

- [x] lingua_franca parse utils
- [x] lingua_franca format utils
- [ ] ovos default skills support that lang
- [x] we have a online STT (high accuracy) option for that lang
- [x] we have a offline STT option for that lang
- [x] we have a online  (high quality) male TTS option for that lang
- [ ] we have a offline male TTS option for that lang
- [x] we have a online  (high quality) female TTS option for that lang
- [x] we have a offline female TTS option for that lang
      
Minimal plugin recommendations:
offline:
   - male TTS - ?
      - espeak sucks but works
   - female TTS - pico
   - STT - vosk
  
online:
   - male TTS -  larynx - es-es/carlfm-glow_tts
   - female TTS - larynx - es-es/karen_savage-glow_tts
   - STT - google (ovos proxy)

**Comment by AIIX** [comment]
German Checklist:

- [x] lingua_franca parse utils
- [x] lingua_franca format utils
- [ ] ovos default skills support that lang
- [x] we have a online STT (high accuracy) option for that lang
- [x] we have a offline STT option for that lang
- [x] we have a online  (high quality) male TTS option for that lang
- [ ] we have a offline male TTS option for that lang
- [x] we have a online  (high quality) female TTS option for that lang
- [x] we have a offline female TTS option for that lang
      
Minimal plugin recommendations:
offline:
   - male TTS - ?
       - espeak works but sucks
   - female TTS - pico
   - STT - vosk
  
online:
   - male TTS - larynx - de-de/thorsten-glow_tts
   - female TTS - larynx - de-de/eva_k-glow_tts
   - STT - google (ovos proxy)

**Reply by domcross** [reply]
> German Checklist:
> 
> * [ ]  we have a offline male TTS option for that lang
> 
> Minimal plugin recommendations: offline:
> 
> * male TTS - ?
>   
>   * espeak works but sucks

Did you consider Mimic3 "de_DE/thorsten_low" (RTF around 0.5 on a RPI4) already?

**Comment by AIIX** [comment]
Italian Checklist:

- [x] lingua_franca parse utils
- [x] lingua_franca format utils
- [ ] ovos default skills support that lang
- [x] we have a online STT (high accuracy) option for that lang
- [x] we have a offline STT option for that lang
- [ ] we have a online  (high quality) male TTS option for that lang
- [ ] we have a offline male TTS option for that lang
- [ ] we have a online  (high quality) female TTS option for that lang
- [x] we have a offline female TTS option for that lang
      
Minimal plugin recommendations:
offline:
   - male TTS - ?
   - female TTS - pico
   - STT - vosk
  
online:
   - male TTS - ?
   - female TTS - ?
   - STT - google (ovos proxy)

**Comment by AIIX** [comment]
Portuguese Checklist:

- [x] lingua_franca parse utils
- [x] lingua_franca format utils
- [ ] ovos default skills support that lang
- [x] we have a online STT (high accuracy) option for that lang
- [x] we have a offline STT option for that lang
- [x] we have a online  (high quality) male TTS option for that lang
- [ ] we have a offline male TTS option for that lang
- [ ] we have a online  (high quality) female TTS option for that lang
- [ ] we have a offline female TTS option for that lang
      
Minimal plugin recommendations:
offline:
   - male TTS - ?
      -  espeak sucks but works
   - female TTS - ?
      - espeak sucks but works
   - STT - vosk (works for pt-br , not very good for pt-pt)
  
online:
   - male TTS - coqui neon model
   - female TTS - ?
       - polly TTS has some good voices, but its not free
   - STT - google (ovos proxy)

**Comment by AIIX** [comment]
Dutch Checklist:

- [x] lingua_franca parse utils
- [x] lingua_franca format utils
- [ ] ovos default skills support that lang
- [x] we have a online STT (high accuracy) option for that lang
- [x] we have a offline STT option for that lang
- [ ] we have a online  (high quality) male TTS option for that lang
- [ ] we have a offline male TTS option for that lang
- [ ] we have a online  (high quality) female TTS option for that lang
- [ ] we have a offline female TTS option for that lang
      
Minimal plugin recommendations:
offline:
   - male TTS - ?
   - female TTS - ?
   - STT - vosk
  
online:
   - male TTS - ?
   - female TTS - ?
   - STT - google (ovos proxy)

**Comment by AIIX** [comment]
please review lang translations on https://github.com/OpenVoiceOS/ovos-core/tree/dev/mycroft/res/ui/translations and  for homescreen on 
https://github.com/OpenVoiceOS/skill-ovos-homescreen/pull/41 for timer on  https://github.com/OpenVoiceOS/skill-ovos-timer/pull/7,  and for weather on https://github.com/OpenVoiceOS/skill-ovos-weather/pull/9
